src/JsonSchema/SchemaFactory.php

```diff
@@ -143,12 +143,18 @@ public function buildSchema(string $className, string $format = 'json', string $
         foreach ($this->propertyNameCollectionFactory->create($inputOrOutputClass, $options) as $propertyName) {
             $propertyMetadata = $this->propertyMetadataFactory->create($inputOrOutputClass, $propertyName, $options);
 
-            if (false === $propertyMetadata->isReadable() && false === $propertyMetadata->isWritable()) {
+            if (
```

Comment:
Looking at the implementation of OpenApiFactory, there is
```
$this->jsonSchemaFactory->buildSchema($resourceClass, $operationFormat, Schema::TYPE_OUTPUT, $operation, $schema, null, $forceSchemaCollection);
```
for the output of an api call (for instance GET)

and
```
$this->jsonSchemaFactory->buildSchema($resourceClass, $operationFormat, Schema::TYPE_INPUT, $operation, $schema, null, $forceSchemaCollection);
```
for the input of an api call PATCH/PUT/POST

Therefore I think
- We should expose in the input schema only Writable properties
- We should expose in the output schema only Readable properties (and declare them required)

It makes no sens to me to add writable property in the GET schema (or the opposite). Am i misunderstandind ?

Reply:
Still it means that we shouldn't use the same schema for input and output and therefor the same cache key.

Should we start with a different cache key without changing anything?